pkg/pod/pod.go

```diff
@@ -167,7 +168,7 @@ func MakePod(images pipeline.Images, taskRun *v1alpha1.TaskRun, taskSpec v1alpha
 	// TODO(#1605): Remove this loop and make each transformation in
 	// isolation.
 	for i, s := range stepContainers {
-		if s.WorkingDir == "" {
+		if s.WorkingDir == "" && shouldOverrideWorkingDir(kubeclient) {
```

Comment:
You should pre-compute this outside the for loop

Reply:
done.